### ISSUE TITLE
weird crash on RPi

### ISSUE THREAD

**Issue by SvenDowideit** [LudovicRousseau/pyscard#71]
first up - thank you for maintaining pyscard! its been working well for the last year on the RaspberryPi's we've been using

but... I did some pretty major changes to the code, and now I'm getting a weird crash:

```
INFO:root:['ACS ACR122U 00 00']
INFO:root:Waiting for smartcard or RFID.
INFO:root:


INFO:root:retry:
INFO:root:+Inserted: 3B8F8001804F0CA000000306030001000000006A
INFO:root:tagid AD6BC255
INFO:root:+Removed: 3B8F8001804F0CA000000306030001000000006A
INFO:root:+Inserted: 3B8F8001804F0CA000000306030001000000006A
INFO:root:tagid AD6BC255
INFO:root:+Inserted: 
Traceback (most recent call last):
  File "/usr/lib/python2.7/dist-packages/smartcard/CardMonitoring.py", line 176, in run
    (addedcards, removedcards))
  File "/usr/lib/python2.7/dist-packages/smartcard/Observer.py", line 63, in notifyObservers
    observer.update(self, arg)
  File "./rfid-mifare/main.py", line 61, in update
    connection.connect( CardConnection.T1_protocol )
  File "/usr/lib/python2.7/dist-packages/smartcard/CardConnectionDecorator.py", line 54, in connect
    self.component.connect(protocol, mode, disposition)
  File "/usr/lib/python2.7/dist-packages/smartcard/pcsc/PCSCCardConnection.py", line 128, in connect
    SCardGetErrorMessage(hresult))
CardConnectionException: Unable to connect with protocol: T1. Card is unpowered.
```

it doesn't happen on my amd64 box :)

I'm going to take the code apart to try to localise it - I _think_ it started happening when I added python logging, but atm, i thought I'd just record this in case it helps someone.

**Comment by SvenDowideit** [comment]
taking out the mqtt network loop seems to stop the crash from happening - I'll keep poking it till i learn something.

**Comment by SvenDowideit** [comment]
it wasn't the logging - removed that and it still happens.

for a while it looked like removing the paho_mqtt network loop stopped it crashing, but that didn't last

**Comment by LudovicRousseau** [comment]
I don't think your problem is with PySCard.
Generate pcscd logs as described in https://pcsclite.apdu.fr/#support

**Comment by LudovicRousseau** [comment]
No feedback. Closing.

**Comment by SvenDowideit** [comment]
sorry, I needed to get the thing released :(
It looks to me like it may be a problem either with PySCard, or quite likely, the way i'm using it.
But - its exacerbated by some PyGame code that was running in a separate process (and the fact i'm running all this on Raspberry Pi's)
It essentially looks like the getuid requests that I made in the update code were delayed (due to system load) just long enough for the rfid tag to not be there anymore.
For this release, we removed the audio process, and its been stable since.

 its not CPU load that causes this - so I presume PyGame has all sorts of responsiveness settings

hopefully I'll circle back to figuring out more details - though its just as likely I may replace PyGame with something less intensive.
